WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -138,12 +138,21 @@ struct BlazeCampaignDashboardView: View {
             campaignDetailView(url: url)
         }
         .sheet(isPresented: $viewModel.shouldShowIntroView) {
-            BlazeCampaignIntroView(onStartCampaign: {
-                viewModel.shouldShowIntroView = false
-                startCampaignFromIntroTapped?(selectedProductID)
-            }, onDismiss: {
-                viewModel.shouldShowIntroView = false
-            })
+            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
+                BlazeCreateCampaignIntroView(onCreateCampaign: {
+                    viewModel.shouldShowIntroView = false
+                    startCampaignFromIntroTapped?(selectedProductID)
+                }, onDismiss: {
+                    viewModel.shouldShowIntroView = false
+                })
+            } else {
+                BlazeCampaignIntroView(onStartCampaign: {
+                    viewModel.shouldShowIntroView = false
+                    startCampaignFromIntroTapped?(selectedProductID)
+                }, onDismiss: {
+                    viewModel.shouldShowIntroView = false
+                })
+            }
```

Comment:
Suggestion: we can move this to a separate method and try to reuse the logic for the handler to avoid duplicating it:

```Swift
var onCreateCampaign: () -> Void = {
    viewModel.shouldShowIntroView = false
    startCampaignFromIntroTapped?(selectedProductID)
}

var onDismissIntroView: () -> Void = {
    viewModel.shouldShowIntroView = false
}

var introView: some View {
    if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
        BlazeCreateCampaignIntroView(onCreateCampaign: onCreateCampaign, onDismiss: onDismiss)
    } else {
        BlazeCampaignIntroView(onCreateCampaign: onCreateCampaign, onDismiss: onDismiss)
    }
}

```

Reply:
(similar suggestions for `ProductFormViewController` and `BlazeCampaignListView`).

Reply:
- The callbacks in `BlazeCampaignListView` and `BlazeCampaignDashboardView` updates view model values and execute the callback `startCampaignFromIntroTapped`. 
- `ProductFormViewController` also works in the same way. 

I don't think adding a separate method and reusing it is straightforward/useful. Please correct me if I misunderstood your suggestion.

Reply:
Thanks for explaining. Addressed this in 3b12a8467cb7dce31adb0a90e03dadce39b94328